pkg/cloud/rhos/rhos.go

```diff
@@ -0,0 +1,204 @@
+package rhos
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gophercloud/gophercloud/openstack"
+	"strconv"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/submariner-io/cloud-prepare/pkg/ocp"
+	"gopkg.in/yaml.v2"
+
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/stolostron/submariner-addon/pkg/constants"
+	"github.com/submariner-io/cloud-prepare/pkg/k8s"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/stolostron/submariner-addon/pkg/cloud/reporter"
+
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	cloudpreparerhos "github.com/submariner-io/cloud-prepare/pkg/rhos"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	cloudsYAMLName = "clouds.yaml"
+	cloudName      = "cloud"
+	gwInstanceType = "PnTAE.CPU_4_Memory_8192_Disk_50"
```

Comment:
I have a feeling that this is not a common flavor that is available across all OCP deployments on OSP, but since @aswinsuryan mentioned earlier that it is what is selected with OCP, we can move ahead. The reason I'm emphasising this again is because currently we are not showing the flavor option to the user, so in the absence of this flavor, cloud-prepare would run into issues.

Reply:
@sridhargaddam To deploy submariner on RHOS through addon, we ned to manually create submarinerconfig. So user can always specify custom flavor when creating Submarinerconfig. By the time we add RHOS to UI, option will be made available.

Reply:
I thought the SubmarinerConfig will be automatically created when user tries to install Submariner Addon on the ManagedCluster running on RHOS. However, we can always **update** it, if required. Is this not the case?

Reply:
It is done by UI for clouds that are supported through UI. for RHOS, vSphere etc. It will at most create ManagedClusterAddon object in that cluster's namespace on hub. Note that this is the UI use case where you add non supported cluster to a clusterset with submariner addon already installed.